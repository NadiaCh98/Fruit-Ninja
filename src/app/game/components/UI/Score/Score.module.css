.wrapper {
  position: absolute;
  z-index: 2;
}

.score {
  grid-area: score;
  font-size: var(--text-size);
  background: var(--yellow-gradient);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
}

.best {
  grid-area: best;
  font-size: calc(var(--text-size) / 2.5);
  background: linear-gradient(
    rgba(255, 179, 0, 1) 0%,
    rgba(255, 111, 0, 1) 100%
  );
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
}
