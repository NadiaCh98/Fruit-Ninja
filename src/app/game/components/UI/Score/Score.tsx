import React, { memo } from 'react';
import { GameControls } from '../../GameControls/GameControls';
import styles from './Score.module.css';

interface ScoreProps {
  readonly score: number;
  readonly best: number;
}

export const Score: React.FC<ScoreProps> = memo(({ score, best }) => {
  return (
    <GameControls>
      <p className={styles.score}>{score}</p>
      <p className={styles.best}>Best: {best}</p>
    </GameControls>
  );
});

Score.displayName = 'Score';
