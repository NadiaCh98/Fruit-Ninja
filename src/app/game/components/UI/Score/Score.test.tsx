import { render } from '@testing-library/react';
import React from 'react';
import { Score } from './Score';

describe('Score', () => {
  const score = <Score score={50} best={60} />;
  it('has a correct template', () => {
    const { container } = render(score);

    expect(container).toMatchSnapshot();
  });
});
