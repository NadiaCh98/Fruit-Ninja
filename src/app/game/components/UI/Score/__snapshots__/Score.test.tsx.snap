// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Score has a correct template 1`] = `
<div>
  <div
    class="sc-bdnxRM lhbDim"
  >
    <p
      class="score"
    >
      50
    </p>
    <p
      class="best"
    >
      Best: 
      60
    </p>
  </div>
</div>
`;
